plugins/chat/assets/javascripts/discourse/models/user-chat-channel-membership.js

```diff
@@ -1,3 +1,30 @@
 import RestModel from "discourse/models/rest";
+import { tracked } from "@glimmer/tracking";
+import User from "discourse/models/user";
+export default class UserChatChannelMembership extends RestModel {
+  @tracked following = false;
+  @tracked muted = false;
+  @tracked unread_count = 0;
+  @tracked unread_mentions = 0;
+  @tracked chat_message_id = null;
+  @tracked chat_channel_id = null;
+  @tracked desktop_notification_level = null;
+  @tracked mobile_notification_level = null;
+  @tracked last_read_message_id = null;
+}
```

Comment:
Untested... but I think you should be able to use `static` rather than `reopenClass`?

```suggestion
  static create(args) {
    args = args || {};
    this._initUser(args);
    return super.create(args);
  }
  
  static _initUser(args){
    if (args.user instanceof User) {
      return;
    }

    args.user = User.create(args.user);
  }
}
```

Reply:
That would make sense yes, will try 👌

Reply:
Couldn't make it work for now, will try more after the merge